### ISSUE TITLE
AbbreviationAsWordInName documentation need to extended

### ISSUE THREAD

**Issue by mcnulty** [checkstyle/checkstyle#10889]
I have read check documentation: https://checkstyle.sourceforge.io/config_naming.html#AbbreviationAsWordInName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac TestABbreviation.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <property name="charset" value="UTF-8"/>

  <property name="severity" value="warning"/>

  <property name="fileExtensions" value="java, properties, xml"/>

  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="ignoreFinal" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="allowedAbbreviations" value="AB"/>
      <property name="tokens"
               value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                    PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF,
          RECORD_COMPONENT_DEF"/>
    </module>
  </module>
</module>

/var/tmp $ cat TestABbreviation.java
public class TestABbreviation {}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.0.1-all.jar -c config.xml TestABbreviation.java
Starting audit...
[WARN] /var/tmp/TestABbreviation.java:1:14: Abbreviation in name 'TestABbreviation' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
Audit done.
```
---

Given that `allowedAbbreviations` is set to `AB`, I would expect the `TestABbreviation` name to be valid. It appears there is an off-by-one error in `AbbreviationAsWordInName` because setting `allowedAbbreviations` to `A` results in no warnings:

```bash
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <property name="charset" value="UTF-8"/>

  <property name="severity" value="warning"/>

  <property name="fileExtensions" value="java, properties, xml"/>

  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="ignoreFinal" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="allowedAbbreviations" value="A"/>
      <property name="tokens"
               value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                    PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF,
          RECORD_COMPONENT_DEF"/>
    </module>
  </module>
</module>

/var/tmp $ java $RUN_LOCALE -jar ./checkstyle-9.0.1-all.jar -c ./config.xml TestABbreviation.java
Starting audit...
Audit done.
```

Interestingly enough, if the abbreviation is at the end of the name, the check works as expected:

```bash
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <property name="charset" value="UTF-8"/>

  <property name="severity" value="warning"/>

  <property name="fileExtensions" value="java, properties, xml"/>

  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="ignoreFinal" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="allowedAbbreviations" value="AB"/>
      <property name="tokens"
               value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                    PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF,
          RECORD_COMPONENT_DEF"/>
    </module>
  </module>
</module>

/var/tmp $ cat TestAB.java
public class TestAB {}

/var/tmp $ java $RUN_LOCALE -jar ./checkstyle-9.0.1-all.jar -c ./config.xml TestABbreviation.java
Starting audit...
Audit done.
```

**Comment by rnveach** [comment]
The documentation should be updated to make this clearer this is how the abbreviation identification works.

**Comment by MUzairS15** [comment]
Hello, everyone  I am new to this project and looking for beginner freindly issue, so can I take this?

**Comment by romani** [comment]
@MUzairS15, please send a PR.

**Comment by MUzairS15** [comment]
When I ran `mvn test jacoco:restore-instrumented-classes jacoco:report@default-report` this is failing with error  `XdocsJavaDocsTest.testAllCheckSectionJavaDocs:167->examineCheckSection:194`.
Should I push my PR (or a Draft PR)?

**Comment by romani** [comment]
Try to understand why it is failing, sounds like you structure of content is damaged a bit

**Comment by MUzairS15** [comment]
I have raised a PR can you please check and share your thoughts?

**Comment by MUzairS15** [comment]
@romani When you are free pls check

**Comment by romani** [comment]
Fix is merged
